src/index.ts

```diff
@@ -1,26 +1,42 @@
 import express, { Request, Response } from 'express';
 import db from './knex';
-//comment
+import cors from 'cors';
+import dotenv from 'dotenv';
+import jwt from 'jsonwebtoken';
+
+dotenv.config();
 const app = express();
 const port = 3000;
 
+const JWT_SECRET = process.env.JWT_SECRET || 'key';
+
 // Middleware
 app.use(express.json());
+app.use(cors({ origin: 'http://localhost:5173' }));
 
-// Basic GET Route
-app.get('/', (req: Request, res: Response) => {
-  res.send('Hello World!');
-});
 
-app.get('/users', async (req: Request, res: Response) => {
+// Creating new users -endpoint
+app.post('/users', async (req: Request, res: Response) => {
+  const { email, password } = req.body;
+
+  if (!email || !password) {
+    return res.status(400).json({ message: 'Email and password are required' });
+  }
+
   try {
-    const users = await db('table-1').select('*');
-    res.json(users);
-  } catch (error) {
-    res.status(500).json({ message: 'Error fetching users', error });
+    const [userId] = await db('users').insert({ email, password }).returning('id');
+    const token = jwt.sign({ userId }, JWT_SECRET, { expiresIn: '1h' });
+    res.status(201).json({ message: 'User created successfully', token, userId });
+  } catch (error: any) {
+    console.error('Error creating user:', error); 
+    if (error.code === 'ER_DUP_ENTRY') {
+      return res.status(400).json({ message: 'Email already exists' });
+    }
+    res.status(500).json({ message: 'Error creating user', error });
```

Comment:
Add the default error handle for express.js (https://expressjs.com/en/guide/error-handling.html, Writing error handlers)

In the default error handler function, check `NODE_ENV` environment value. If the value is different then `production`, you can return the `error` variable to the user. Otherwise, just return `An error occurred` message.

After the default error handler, we don't need to use try-catch block.